src/agent/src/sandbox.rs

```diff
@@ -403,6 +417,124 @@ impl Sandbox {
+                // A process can't join a new mount namespace if it is sharing
+                // filesystem-related attributes (using CLONE_FS flag) with another process.
+                // Ref: https://man7.org/linux/man-pages/man2/setns.2.html
+                //
+                // The implementation of the Rust standard library's std::thread relies on
+                // the CLONE_FS parameter at the low level.
+                // Therefore, it is not possible to switch directly to the mount namespace using setns.
+                // Instead, it is necessary to first switch to a new mount namespace using unshare.
+                unshare(CloneFlags::CLONE_NEWNS)?;
+                for m in mounts {
+                    if let Some(src_init_pid) = src_ctrs.get(m.src_ctr()) {
+                        // Shared mount points are created by applications within the source container,
+                        // so we need to ensure they are already prepared.
+                        setns(init_mntns, CloneFlags::CLONE_NEWNS)?;
+                        loop {
+                            let mut is_ready = false;
+                            let proc_mounts_path = format!("/proc/{}/mounts", *src_init_pid);
+                            let proc_mounts = fs::read_to_string(proc_mounts_path.as_str())?;
+                            let lines: Vec<&str> = proc_mounts.split('\n').collect();
+                            for line in lines {
+                                let parts: Vec<&str> = line.split_whitespace().collect();
+                                if parts[1] == m.src_path() {
+                                    is_ready = true;
+                                    break;
+                                }
+                            }
+
+                            if is_ready {
+                                break;
+                            }
+
+                            thread::sleep(Duration::from_millis(100));
+                        }
+
+                        // Switch to the src container to obtain shared mount points.
+                        let src_mntns_path = format!("/proc/{}/ns/mnt", *src_init_pid);
+                        let src_mntns = fcntl::open(
+                            src_mntns_path.as_str(),
+                            OFlag::O_RDONLY | OFlag::O_CLOEXEC,
+                            Mode::empty(),
+                        )?;
+                        setns(src_mntns, CloneFlags::CLONE_NEWNS)?;
+                        let src = std::ffi::CString::new(m.src_path())?;
+                        let mount_fd = unsafe {
+                            syscall(
+                                libc::SYS_open_tree,
+                                libc::AT_FDCWD,
+                                src.as_ptr(),
+                                0x1 | 0x8000 | libc::O_CLOEXEC, // OPEN_TREE_CLONE | AT_RECURSIVE | OPEN_TREE_CLOEXEC
+                            ) as i32
+                        };
```

Comment:
The same applies to `src_mntns` and `mount_fd` as well. Please make sure that they are closed properly.